services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/storage.py

```diff
@@ -15,30 +16,53 @@
 _LIVENESS_TIMEOUT: Final[timedelta] = timedelta(seconds=5)
 
 
-async def _is_storage_responsive(url: str) -> bool:
+def _get_auth(storage_auth_settings: StorageAuthSettings) -> tuple[str, str] | None:
+    if storage_auth_settings.auth_required:
+        assert storage_auth_settings.STORAGE_USERNAME  # nosec
+        assert storage_auth_settings.STORAGE_PASSWORD  # nosec
+        return (
+            storage_auth_settings.STORAGE_USERNAME,
+            storage_auth_settings.STORAGE_PASSWORD.get_secret_value(),
+        )
+    return None
+
+
+def _get_url(storage_auth_settings: StorageAuthSettings) -> str:
+    url = URL(storage_auth_settings.api_base_url).with_path("/")
+    return f"{url}"
+
+
+async def _is_storage_responsive(storage_auth_settings: StorageAuthSettings) -> bool:
+    url = _get_url(storage_auth_settings)
+    auth = _get_auth(storage_auth_settings)
```

Comment:
Why not adding it as a member of `StorageAuthSettings` as ``get_auth_or_none() -> AuthTuple | None`` that should return AuthTuple if `auth_required` or `None` otherwise?

Reply:
Because the "Auth" format depends on the client. We have both httpx and aiohttp clients. And they both have different formats.

Reply:
Fine but you can always map from the named tuple, right? I think here the point is that it is better to have **named** tuples that just tuples.